File: C#Part2 - Homework/03. Methods - Homework/06.FirstLargerThanNeighbours/FirstLargerThanNeighbours.cs
namespace FirstLargerThanNeighbours
{
    using System;

    class FirstLargerThanNeighbours
    {
        /* Write a method that returns the index of the first element in array that is larger 
           than its neighbours, or -1, if there’s no such element.
           Use the method from the previous exercise.
         */

        static void Main()
        {
            Console.Write("Enter size of array = ");
            int size = int.Parse(Console.ReadLine());
            int[] numbers = new int[size];
            for (int i = 0; i < size; i++)
            {
                Console.Write("Index [{0}] = ", i);
                numbers[i] = int.Parse(Console.ReadLine());
            }
            int index = LargeNumber(numbers);
            Console.WriteLine(index);
        }
        static int LargeNumber(int[] numbs)
        {
            for (int i = 1; i < numbs.Length; i++)
            {
                if (numbs[i - 1] < numbs[i] && numbs[i] > numbs[i + 1])
                {
                    return i;
                }
            }

            return -1;
        }
    }
}

File: JavaScript/Homework/Functions - Homework/Problem 7 First larger than neighbours/FirstLargerThanNeighbours.js
// Write a function that returns the index of the first element in array that is larger than its neighbours or -1, if there’s no such element.
// Use the function from the previous exercise.

var btn = document.getElementById('print-nums-button');

function checkIfNumIsBigger(arr, position) {
    var isBigger = false;
    position = position - '0';
    if (((position - 1) > 0) && ((position + 1) < arr.length)) {
        if (arr[position] > arr[position - 1] && arr[position] > arr[position + 1])
            isBigger = true;
    } else if (position - 1 <= 0) {
        if (arr[position] > arr[position + 1])
            isBigger = true;
    }
    else if (position + 1 >= arr.length) {
        if (arr[position] > arr[position - 1])
            isBigger = true;
    }

    return isBigger;
}
function findFirstBigger(arr) {
    var position = -1;
    for (var i = 0; i < arr.length; i++) {
        if (checkIfNumIsBigger(arr, i)) {
            position = i;
            break;
        }
    };
    return position;
}
btn.addEventListener('click', function () {
    var inputValue = document.getElementById('value').value;
    var resultArea = document.getElementById('result');
    var arrayOfDigits = inputValue.split(' ');
    var position = findFirstBigger(arrayOfDigits);
    result.innerHTML = "Position: " + position;
})